base58.py

```diff
@@ -73,9 +73,13 @@ def b58decode_int(v):
 def b58decode(v):
     '''Decode a Base58 encoded string'''
 
-    if not isinstance(v, str):
+    if isinstance(v, bytes):
```

Comment:
This confusing conditional is because of python2, where bytes and str is the same thing. decoding the python2 str will give a unicode object which later breaks things.